client/src/test/java/io/pravega/client/segment/impl/SegmentInputStreamFactoryImplTest.java

```diff
@@ -90,4 +98,48 @@ public void testCreateInputStreamForSegmentWithOffsetAndBufferSize() {
         assertEquals(Segment.fromScopedName("scope/stream/0"), reader.getSegmentId());
 
     }
+
+    @Test
+    public void testGetDelegationTokenProviderWithCacheHitAndNotExpire() {
+        long segmentId = 1L;
+        MockConnectionFactoryImpl connectionFactory = new MockConnectionFactoryImpl();
+        MockController mockController = new MockController("localhost", -1, connectionFactory, false);
+        SegmentInputStreamFactoryImpl inputStreamFactory = spy(new SegmentInputStreamFactoryImpl(mockController, connectionFactory));
+        doReturn(Duration.ofMinutes(1)).when(inputStreamFactory).getExpirationTimeMinute();
+        doReturn(3).when(inputStreamFactory).getCacheMaxSize();
+        SimpleCache<String, DelegationTokenProvider> tokenProviderSimpleCache = inputStreamFactory.getTokenProviderCache();
+        DelegationTokenProvider mockDelegationTokenProvider = mock(DelegationTokenProvider.class);
+        tokenProviderSimpleCache.put(Segment.fromScopedName("testScope/testStream/" + segmentId).getScopedName(), mockDelegationTokenProvider);
+
+        CompletableFuture<String> tokenFuture = new CompletableFuture<>();
+        tokenFuture.complete("testToken");
+        doReturn(tokenFuture).when(mockDelegationTokenProvider).retrieveToken();
+        DelegationTokenProvider delegationTokenProvider = inputStreamFactory.getDelegationTokenProvider(Segment.fromScopedName("testScope/testStream/" + segmentId));
+
+        Assert.assertSame(delegationTokenProvider, mockDelegationTokenProvider);
+    }
+
+    @Test
+    public void testGetDelegationTokenProviderWithCacheHitButExpire() throws InterruptedException {
+        long segmentId = 1L;
+        MockConnectionFactoryImpl connectionFactory = new MockConnectionFactoryImpl();
+        MockController mockController = new MockController("localhost", -1, connectionFactory, false);
+        SegmentInputStreamFactoryImpl inputStreamFactory = spy(new SegmentInputStreamFactoryImpl(mockController, connectionFactory));
+        doReturn(Duration.ofMinutes(1)).when(inputStreamFactory).getExpirationTimeMinute();
+        doReturn(3).when(inputStreamFactory).getCacheMaxSize();
+        SimpleCache<String, DelegationTokenProvider> tokenProviderSimpleCache = inputStreamFactory.getTokenProviderCache();
+        DelegationTokenProvider mockDelegationTokenProvider = mock(DelegationTokenProvider.class);
+        tokenProviderSimpleCache.put("1", mockDelegationTokenProvider);
+
+        Thread.sleep(60000);
```

Comment:
Sleep is not allowed in unit tests. We run over 800 tests and they complete in 13 minutes. Please insure that your tests do not take disproportionate time. They should run in a few milliseconds. To do this we need all tests operated based on time to use a `Supplier<Long>` instead of just using `systemTimeMillis` that was an alternate implementation can be provided, or alternatively find a way to use a mock or have test methods to make the test deterministic without relying on time.